### PR TITLE
refactor(issue-360): used joi to validate developer request body

### PR DESCRIPTION
**Pull request by kenzdozz**
solves Issue #360

### CODE REVIEW

**Review by ijemmao — APPROVED** [review]
Amazing, the validation file looks so much more cohesive

File: src/middleware/validateDeveloperBody.js [hunk 1]
```diff
@@ -1,19 +1,17 @@
...
-  }
+export default (req, res, next) => {
+  const { body: data } = req;
 
-  if (!data.password) {
-    res.status(400);
-    return res.send({ error: 'Password is required' });
+  const validationResult = developersJoiSchema.validate(data);
+  if (validationResult.error) {
+    return res.send({ error: validationResult.error.message });
```
Comment (ijemmao) [inline src/middleware/validateDeveloperBody.js:14]: You need to send the status code of  `400`

**Review by ijemmao — CHANGES_REQUESTED** [review]
Just need to specify the status code

**Comment by kenzdozz** [comment]
> Just need to specify the status code

Yeah, I missed that. Done now

**Review by ijemmao — APPROVED** [review]

**Pull request merged**